Fixed Error In Forming RDM

JoeyT1994 opened the pull request:
Now account for the fact combiners is a directed data graph.

mtfishman commented:
Glad to see this part can be simplified. Could you add a test for this function?

JoeyT1994 commented on src/beliefpropagation.jl:
```diff
@@ -289,14 +289,14 @@ function two_site_rdm_bp(
   psi1 = deepcopy(psi[v1])*prime!(dag(deepcopy(psi[v1])))
   for v in neighbors(psi, v1)
-    C = haskey(combiners,NamedEdge(v => v1)) ? combiners[NamedEdge(v => v1)] : combiners[NamedEdge(v1 => v)]
+    C = combiners[NamedEdge(v => v1)]
```
Okay I will commit that in.

JoeyT1994 commented:
> Glad to see this part can be simplified. Could you add a test for this function?

I have added a testing unit for this now!

mtfishman commented:
Would Metis work instead of KaHyPar for the test? Unfortunately it is a bit awkward getting the automatic testing working with KaHyPar because of the installation issues on Windows.

JoeyT1994 commented:
> 

Good idea, I have changed it and the tests appear to be passing with Metis so I will use that for the testing.

mtfishman commented:
Could you add the line:
```julia
LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
```
to `test/Project.toml`? That will allow the tests to automatically install the `LinearAlgebra` package.

JoeyT1994 commented:
> ```julia
> LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
> ```

JoeyT1994 commented:
Added

> Could you add the line:
> 
> ```julia
> LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
> ```
> 
> to `test/Project.toml`? That will allow the tests to automatically install the `LinearAlgebra` package.

Sure, I added this.

mtfishman commented:
Thanks!
